src/odemis/driver/smaract.py

```diff
@@ -3699,26 +3926,38 @@ def _wait_for_event(self, end_state, timeout=float("inf")):
         timeout (float): maximum time to wait in s. If inf, it will wait forever.
         """
         if timeout == float("inf"):
-            tend = None
-        else:
-            tend = time.time() + timeout
+            timeout = 600  # 10 minutes
+
+        tend = time.time() + timeout
         ev = SA_SI_Event()
 
         state = None
         while state != end_state:
-            if tend:
-                t = int((tend - time.time()) * 1000)  # SA_SI_WaitForEvent accepts timeout in ms
-                if t < 0 or ev.error == SA_SI_ERROR_TIMEOUT:
-                    raise TimeoutError()
-                t = c_uint(t)
-            else:
-                t = SA_SI_TIMEOUT_INFINITE
+            if state is not None:
+                logging.debug("Skipped event 0x%x" % ev.type)
+
+            t = int((tend - time.time()) * 1000)  # SA_SI_WaitForEvent accepts timeout in ms
+            if t < 0:
+                raise TimeoutError("Timeout limit of %s s exceeded." % timeout)
+            t = c_uint(t)
 
             # Don't check event type, it's not always accurate.
             # Instead, only one event is active and all other events are disabled, so it is clear
             # which event we are waiting for.
             self.core.SA_SI_WaitForEvent(self._id, byref(ev), t)
             state = ev.devEventParameter & 0xffff
+            if ev.error != SA_SI_ERROR_NONE:
```

Comment:
I think you got confused by the use of union. Check the section 7 of the programmers' guide.
*Only* if the ev.type is `SA_SI_STREAM_ABORTED_EVENT`, then ev.error is meaningful.
For the SA_PS_AF_ADJUSTMENT_PROGRESS_EVENT, the state is lowest 16-bits of the devEventParameter. That's really specific of this event.
I'd suggest to handle each event type explicitly (with a generic else where you log of the event type which is not handled).

Right now, your function is tailored to wait for the ADJUSTMENT_PROGRESS_EVENT, so you could even rename it `wait_for_progress_event()`. In this case, I would highly suggest to move the small part corresponding to wrapping `SA_SI_WaitForEvent()` to a separate function, which converts the timeout, and return the SA_SI_Event received, or raise a TimeoutError.